src/threads/handlers.spec.ts

```diff
@@ -0,0 +1,56 @@
+import * as parsers from '../parsers';
+import { openThread } from './handlers';
+import { forumTable } from './table';
+
+const mockEvent = (): any => ({});
+const mockContext = (): any => ({});
+const mockCallback = (): any => (() => {});
+
+describe('threads', () => {
+  const getTokenDataStub = jest.spyOn(parsers, 'extractTokenData');
+  const addThreadEventStub = jest.spyOn(forumTable, 'addThreadEvent');
+  addThreadEventStub.mockReturnValue(Promise.resolve());
+
+  const tokenData = (n: number): parsers.TokenData => ({
+    participantId: `openThreadParticipantId-${n}`,
+    itemId: `openThreadItemId-${n}`,
+    userId: `openThreadUserId-${n}`,
+    isMine: true,
+    canWatchParticipant: true,
+  });
+
+  describe('open thread', () => {
+    it('should fail (401) when token data is invalid', async () => {
+      getTokenDataStub.mockReturnValueOnce(new Error('...'));
+      await expect(openThread(mockEvent(), mockContext(), mockCallback())).resolves.toEqual({
+        statusCode: 401,
+        body: '',
+      });
+    });
+
+    it('should succeed (201) when token data is valid', async () => {
+      getTokenDataStub.mockReturnValueOnce(tokenData(1));
+      await expect(openThread(mockEvent(), mockContext(), mockCallback())).resolves.toEqual({
+        statusCode: 201,
+        body: '',
+      });
+    });
+
+    it('should fail gracefully when adding thread opened event fails', async () => {
+      getTokenDataStub.mockReturnValueOnce(tokenData(2));
+      addThreadEventStub.mockReturnValueOnce(Promise.reject(new Error('...')));
+      await expect(openThread(mockEvent(), mockContext(), mockCallback())).rejects.toBeInstanceOf(Error);
+    });
+
+    it('should add an event "thread_opened" to the forum table', async () => {
+      const data = tokenData(3);
+      getTokenDataStub.mockReturnValueOnce(data);
+      await openThread(mockEvent(), mockContext(), mockCallback());
+      expect(addThreadEventStub).toHaveBeenLastCalledWith(
+        data.participantId,
+        data.itemId,
+        { type: 'thread_opened', byUserId: data.userId },
```

Comment:
why isn't it done in the second test case?

Reply:
Fixed